### PR TITLE
Telemetry coverage extension

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Extended coverage for telemetry.

### Reason for changes

- Telemetry update.

### Related tickets

- 154833

### Tests

- TBD

New _categories/events_ are introduced:
| Method  | Event category | Event name | Event label |
| ------------- | ------------- | ------------- | ------------- |
| _nncf.strip_ | nncf_tf,nncf_pt,nncf_pt_fx,nncf_onnx,nncf_ov | _function_call_ | _nncf.strip_ |
| _nncf.data....generate_text_data_ | _nncf_ | _function_call_ | _nncf.data.generate_text_data_ |
| _nncf.torch....wrap_model_ | nncf_pt | _function_call_ | _nncf.torch.wrap_model_ |
| _nncf.torch....load_from_config_ | nncf_pt | _function_call_ | _nncf.torch.load_from_config_ |
| _nncf.torch....get_config_ | nncf_pt | _function_call_ | _nncf.torch.nncf_network.NNCFNetwork.get_config_ |

### CODE REVIEW

File: nncf/data/generators.py [hunk 1]
```diff
@@ -13,13 +13,22 @@
...
 TTokenizer = TypeVar("TTokenizer")
 
 
+@tracked_function(
+    category=NNCF_COMMON_CATEGORY,
+    extractors=[
+        DatasetGeneratedFromApi(),
+    ],
+)
 def generate_text_data(
```
Comment (alexsu52) [inline nncf/data/generators.py:32]: Please mark this function with the api decorator.

File: nncf/common/telemetry_extractors.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+from typing import Any
+
+from nncf.telemetry.extractors import CollectedEvent
+from nncf.telemetry.extractors import TelemetryExtractor
+
+
+class ModelProcessedWithStripApi(TelemetryExtractor):
+    def extract(self, _: Any) -> CollectedEvent:
+        return CollectedEvent(name="model_processed", data="strip_api")
```
Comment (alexsu52) [inline nncf/common/telemetry_extractors.py:20]: @KodiaqQ, @MaximProshin, @AlexKoff88 I would suggest to introduce `function_call` event to cover calling all NNCF functions as following:
```
return CollectedEvent(name="function_call", data="nncf.strip")
```
I would like to know your opinion.

File: nncf/telemetry/events.py [hunk 3]
```diff
@@ -15,6 +15,9 @@
 from nncf.common.utils.backend import BackendType
 from nncf.common.utils.backend import get_backend
 
+# General categories
+NNCF_COMMON_CATEGORY = "nncf_common"
```
[the remark this answers is not in the document]
Comment (KodiaqQ) [reply nncf/telemetry/events.py:19]: It does make no sense in terms of the categorisation. For example, `nncf_ov` or `nncf_onnx` may be the subsets of the `nncf` category, but not `nncf_common`. The originally proposed category reflects the category's differences.

File: nncf/common/telemetry_extractors.py (hunk 2 above)
Comment (KodiaqQ) [reply nncf/common/telemetry_extractors.py:20]: Sounds reasonable to me. 
In this case, we'll have one event with different data values.
The only question is whether it is possible to analyse such an event in the front end.

Comment (alexsu52) [reply nncf/common/telemetry_extractors.py:20]: > Sounds reasonable to me. In this case, we'll have one event with different data values. The only question is whether it is possible to analyse such an event in the front end.

@popovaan, could you provide your opinion?

Comment (popovaan) [reply nncf/common/telemetry_extractors.py:20]: I see no problems in usage of common `function_call` event name and it is convenient in terms of making a report, as all values can be collected using only one filter `Event name`==`function_call`.

If you want a separate group of function calls, which are needed to be a separate table in report (for example, function calls that raised error), you can introduce another event name to group them. For example: failed_function_call. Otherwise separate names per method are redundant.

Comment (KodiaqQ) [reply nncf/common/telemetry_extractors.py:20]: @alexsu52 what should we do here then?

File: nncf/common/strip.py [hunk 4]
```diff
@@ -13,14 +13,18 @@
...
 from nncf.common.utils.backend import BackendType
 from nncf.common.utils.backend import get_backend
+from nncf.telemetry.decorator import tracked_function
+from nncf.telemetry.events import NNCF_COMMON_CATEGORY
 
 TModel = TypeVar("TModel")
 
 
 @api(canonical_alias="nncf.strip")
+@tracked_function(category=NNCF_COMMON_CATEGORY, extractors=[ModelProcessedWithStripApi()])
```
[the remark this answers is not in the document]
Comment (KodiaqQ) [reply nncf/common/strip.py]: Updated.

File: nncf/telemetry/events.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (KodiaqQ) [reply nncf/telemetry/events.py]: Updated.

File: nncf/common/telemetry_extractors.py (hunk 2 above)
Comment (KodiaqQ) [reply nncf/common/telemetry_extractors.py]: Updated.

File: nncf/data/generators.py (hunk 1 above)
Comment (KodiaqQ) [reply nncf/data/generators.py:34]: Updated.

**Comment by KodiaqQ** [comment]
Do we need this, @alexsu52, @MaximProshin?

**Review by alexsu52 — APPROVED** [review]
LGTM

**Review by MaximProshin — APPROVED** [review]
LGTM
